tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -0,0 +1,771 @@
+		Expect(err).ToNot(HaveOccurred(), "Fail to list master nodes")
+		Expect(len(masterNodeList)).To(BeNumerically(">", 0),
+			"Failed to detect master nodes")
+	})
+
+	Context("IBGP Single hop", func() {
+
+		var (
+			nodeAddrList []string
+			addressPool  []string
+			err          error
+			frrk8sPods   []*pod.Builder
+		)
+
+		BeforeAll(func() {
+			By("Setting test iteration parameters")
+			_, _, _, nodeAddrList, addressPool, _, err =
+				metallbenv.DefineIterationParams(
+					ipv4metalLbIPList, ipv6metalLbIPList, ipv4NodeAddrList, ipv6NodeAddrList, "IPv4")
+			Expect(err).ToNot(HaveOccurred(), "Fail to set iteration parameters")
+
+		})
+
+		AfterEach(func() {
+			By("Clean metallb operator and test namespaces")
+			resetOperatorAndTestNS()
+		})
+
+		It("Verify that prefixes configured with alwaysBlock are not received by the FRR speakers",
+			reportxml.ID("74270"), func() {
+				By("Creating a new instance of MetalLB Speakers on workers blocking specific incoming prefixes")
+				err := createNewMetalLbDaemonSetAndWaitUntilItsRunningWithAlwaysBlock(tsparams.DefaultTimeout,
+					workerLabelMap, []string{externalAdvertisedIPv4Routes[0]})
+				Expect(err).ToNot(HaveOccurred(), "Failed to recreate metalLb daemonset")
+
+				By("Creating external FRR pod on master node")
+				frrPod := deployTestPods(addressPool, hubIPaddresses, externalAdvertisedIPv4Routes,
+					externalAdvertisedIPv6Routes)
+
+				By("Creating BGP Peers")
+				createBGPPeerAndVerifyIfItsReady(ipv4metalLbIPList[0], "", 64500,
+					false, frrk8sPods)
+
+				By("Checking that BGP session is established and up")
+				verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+				By("Validating BGP route prefix")
+				validatePrefix(frrPod, "IPv4", removePrefixFromIPList(nodeAddrList), addressPool, 32)
+
+				By("Create a frrconfiguration allow all")
+				err = createFrrConfiguration("frrconfig-allow-all", ipv4metalLbIPList[0],
+					64500, nil, false)
+				Expect(err).ToNot(HaveOccurred(), "Fail to create a FRRconfiguration with allow all")
+
+				frrk8sPods, err := pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+					LabelSelector: "app=frr-k8s",
+				})
+				Expect(err).ToNot(HaveOccurred(), "Fail to find Frrk8 pod list")
+
+				By("Verify that the node FRR pods advertises two routes")
+				err = verifyExternalAdvertisedRoutes(frrPod, ipv4NodeAddrList, externalAdvertisedIPv4Routes)
+				Expect(err).ToNot(HaveOccurred(), "Fail to find advertised routes")
+
+				By("Validate BGP received routes")
+				Eventually(func() string {
+					// Get the routes
+					routes, err := frr.VerifyBGPReceivedRoutesOnFrrNodes(frrk8sPods)
+					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
+
+					// Return the routes to be checked
+					return routes
+				}, 60*time.Second, 5*time.Second).Should(ContainSubstring(externalAdvertisedIPv4Routes[1]),
+					"Fail to find received routes")
+			})
```

Comment:
iiuc this checks that the 2nd prefix is there, but does not validate that it is the only one (i.e that the 1st prefix is    actually blocked), I think this should validate that the `routes` is exactly the 1st prefix (or at least that it does not contain the first prefix)?

Reply:
added verifiying that the blocked route is not received